tests/test_events/test_events.py

```diff
@@ -1598,6 +1598,91 @@ def test_archive_with_exists_event_filter():
     response["EventCount"].should.equal(2)
 
 
+@mock_events
+def test_archive_with_prefix_event_filter():
+    client = boto3.client("events", "eu-central-1")
+    event_bus_arn = "arn:aws:events:eu-central-1:{}:event-bus/default".format(
+        ACCOUNT_ID
+    )
+    client.create_archive(
+        ArchiveName="with-prefix-filter",
+        EventSourceArn=event_bus_arn,
+        EventPattern=json.dumps({"detail": {"foo": [{"prefix": "bar"}]}}),
+    )
+    matching_event_1 = {"Source": "", "DetailType": "", "Detail": '{"foo": "bar"}'}
+    matching_event_2 = {"Source": "", "DetailType": "", "Detail": '{"foo": "bar!"}'}
+    not_matching_event = {"Source": "", "DetailType": "", "Detail": '{"foo": "ba"}'}
+    response = client.put_events(
+        Entries=[matching_event_1, matching_event_2, not_matching_event]
+    )
+    response["FailedEntryCount"].should.equal(0)
+    response = client.describe_archive(ArchiveName="with-prefix-filter")
+    response["EventCount"].should.equal(2)
```

Comment:
Can we extend the assertions here (for all tests) to verify that it returns the correct events? That would make the tests more robust

Reply:
Shall do. I was wondering whether it'd be worth pulling the logic out into an `EventPattern` class and testing the filtering method directly, rather than doing so through an `Archive`. I think the same logic is common to `EventPattern`s on `Rule`s, too.

Reply:
Makes sense, if we can reuse the logic anyway